be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -105,6 +105,336 @@ private:
+            }
+        } else {
+            offset_column = down_cast<Int64Column*>(
+                    ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[1]).get());
+        }
+
+        Int64Column* length_column = nullptr;
+        if (columns[2]->is_nullable()) {
+            is_nullable = true;
+            has_null = (columns[2]->has_null() || has_null);
+
+            const auto* src_nullable_column = down_cast<const NullableColumn*>(columns[2].get());
+            length_column = down_cast<Int64Column*>(src_nullable_column->data_column().get());
+            if (null_result) {
+                null_result = FunctionHelper::union_null_column(null_result, src_nullable_column->null_column());
+            } else {
+                null_result = NullColumn::create(*src_nullable_column->null_column());
+            }
+        } else {
+            length_column = down_cast<Int64Column*>(
+                    ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[2]).get());
+        }
+
+        ArrayColumn* dest_data_column = nullptr;
+        if (columns[0]->is_nullable()) {
+            auto& dest_nullable_column = down_cast<NullableColumn&>(*dest_column);
+            dest_data_column = down_cast<ArrayColumn*>(dest_nullable_column.data_column().get());
+            auto& dest_null_data = dest_nullable_column.null_column_data();
+
+            dest_null_data = null_result->get_data();
+            dest_nullable_column.set_has_null(has_null);
+        } else {
+            dest_data_column = down_cast<ArrayColumn*>(dest_column.get());
+        }
+
+        for (size_t i = 0; i < chunk_size; i++) {
+            _array_slice_item(array_column, i, dest_data_column, offset_column->get(i).get_int64(),
+                              length_column->get(i).get_int64());
+        }
+
+        if (is_nullable) {
+            if (columns[0]->is_nullable()) {
+                return dest_column;
+            } else {
+                return NullableColumn::create(dest_column, null_result);
+            }
+        } else {
+            return dest_column;
+        }
+    }
+
+    static void _array_slice_item(ArrayColumn* column, size_t index, ArrayColumn* dest_column, int64_t offset,
+                                  int64_t length) {
+        auto& dest_offsets = dest_column->offsets_column()->get_data();
+        if (!offset) {
+            dest_offsets.emplace_back(dest_offsets.back());
+            return;
+        }
+
+        Datum v = column->get(index);
+        const auto& items = v.get<DatumArray>();
+
+        if (offset > 0) {
+            // because offset start with 1.
+            --offset;
+        } else {
+            offset += items.size();
+        }
+
+        auto& dest_data_column = dest_column->elements_column();
+        auto end = std::max((int64_t)0, std::min((int64_t)items.size(), (offset + length)));
+        offset = (offset > 0 ? offset : 0);
```

Comment:
what's the behavior of offset < 0?

Reply:
what's the behavior of length < 0?

Reply:
It start at last element, and follow opposite direct,